### PR TITLE
Add support for multiple SMS retrievals

### PR DESCRIPTION
**Pull request by jmrossy**
Hi Bruno,

My team and I needed the ability to retrieve multiple messages using the SMS Retriever api so we added support for that to your nice module :)

If you'd like to merge part of all of those changes upstream then feel free.

### CODE REVIEW

File: android/src/main/java/me/furtado/smsretriever/SmsBroadcastReceiver.java [hunk 1]
```diff
@@ -6,79 +6,69 @@
...
-                }
-            }
+        if (!SmsRetriever.SMS_RETRIEVED_ACTION.equals(intent.getAction())) {
+            return;
         }
-    }
-
-    //region - Privates
 
-    private void emitJSEvent(@NonNull final String key, final String message) {
```
Comment (jmrossy) [inline android/src/main/java/me/furtado/smsretriever/SmsBroadcastReceiver.java]: Moved to SmsHelper to keep this broadcast listener simple

**Comment by Bruno-Furtado** [comment]
Hey @jmrossy, really nice!
Just out of curiosity, why do you need to intercept multiple messages?

**Comment by jmrossy** [comment]
> Just out of curiosity, why do you need to intercept multiple messages?

@Bruno-Furtado Our app requires codes from 3 sms messages as part of our identify verification process.

If you think other apps may need this capability, I'm happy to work with you to get it merged (feel free to suggest changes). 
Otherwise, I understand if you'd rather keep the module's interface simple :)

**Comment by Bruno-Furtado** [comment]
@jmrossy, cool! Until the end of this week, I will take a look. 🙂
Can you check the [codebeat issues](https://codebeat.co/projects/github-com-bruno-furtado-react-native-sms-retriever-master/pull_requests/214622)?

**Comment by Bruno-Furtado** [comment]
@jmrossy, any feedback?

**Comment by jmrossy** [comment]
@Bruno-Furtado not yet, forgot to have a look. Hopefully I'll find time to address those soon :)

**Comment by Bruno-Furtado** [comment]
@jmrossy I will close it since there is some codebeat issues.

**Pull request closed without merging**